spec/v2/providers/storage.spec.ts

```diff
@@ -0,0 +1,336 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import * as config from '../../../src/config';
+import * as options from '../../../src/v2/options';
+import * as storage from '../../../src/v2/providers/storage';
+import { FULL_OPTIONS, FULL_TRIGGER } from './helpers';
+
+describe('v2/storage', () => {
+  describe('_getOptsAndBucket', () => {
+    it('should return throw error without a bucket', () => {
+      expect(() => storage._getOptsAndBucket()).to.throw();
+    });
+
+    it('should return the default bucket on no params', () => {
+      const configStub = sinon
+        .stub(config, 'firebaseConfig')
+        .returns({ storageBucket: 'default-bucket' });
+
+      const [opts, bucket] = storage._getOptsAndBucket();
+
+      configStub.restore();
+      expect(opts).to.deep.equal({});
+      expect(bucket).to.eq('default-bucket');
+    });
+
+    it('should return the given bucket', () => {
+      const [opts, bucket] = storage._getOptsAndBucket('my-bucket');
+
+      expect(opts).to.deep.equal({});
+      expect(bucket).to.eq('my-bucket');
+    });
+
+    it('should return the given bucket and opts', () => {
+      const [opts, bucket] = storage._getOptsAndBucket({
+        bucket: 'my-bucket',
+        region: 'us-west1',
+      });
+
+      expect(opts).to.deep.equal({ region: 'us-west1' });
+      expect(bucket).to.eq('my-bucket');
+    });
+  });
+
+  describe('_onOperation', () => {
+    const EVENT_TRIGGER = {
+      eventType: 'google.cloud.storage.event-type',
+      resource: 'my-bucket',
+      service: 'storage.googleapis.com',
+    };
```

Comment:
nit* I'd make this module-level constant instead of scoping it to this test.

Reply:
Moved to the top and added in specific event constants for each event type.